comfy/node_cache.py

```diff
@@ -0,0 +1,72 @@
+from collections import OrderedDict, namedtuple
+from functools import wraps
+import os
+
+import torch
+
+MAX_CHECKPOINT_CACHE = int(os.environ.get("MAX_CHECKPOINT_CACHE", 1))
+MAX_CONTROLNET_CACHE = int(os.environ.get("MAX_CONTROLNET_CACHE", 1))
+MAX_VAE_CACHE = int(os.environ.get("MAX_VAE_CACHE", 1))
+
+
+class Cache:
+    def __init__(self, max_cache_size=0):
+        self._max_size = max_cache_size
+        self._cache = OrderedDict()
+
+    @property
+    def cache(self):
+        return self._cache
+
+    @property
+    def size(self):
+        return self._max_size
+
+    def clear_cache(self):
+        while len(self._cache) > self.size:
+            k, _ = self._cache.popitem(last=False)
+            print(f"[==Cache==]  Removed {k} from cache.")
+
+    def add_cache(self, k, item):
+        """模型patch会修改key的名字，缓存之前保留一份key的数据和值的引用"""
```

Comment:
Should keep comments in English for PRs to the main repo

Reply:
Sorry for that, I ignored this. So I made second commit to fix this comment, b think I shoud make another PR to do this right?

Reply:
Sorry about that, I fixed already.